### DISCUSSION TITLE
[UX] Report having used the add-on some time in practise

### DISCUSSION THREAD

**Discussion by rugk** [rugk/unicodify#34] [Show and tell]
So to see what are major blockers for a complete v1.0 release, I've used the AMO version (v0.1) for some time. Here is what I experienced:

I'd link the issues with concrete fix requests. Or let's open new ones if we need to…

## Twitter, see https://github.com/rugk/unicodify/issues/4

It's quite funny actually everytime I try to write a tweet I get complete garbage and think I'm writing in a fantasy language or Twitter wants to troll me and then it takes some seconds until I actually remember this is caused by this add-on here. I then need to disable it, write my tweet and then re-enable it.
I am actually not that often on Twitter though.

Though, if I were not the developer of this add-on, I would likely only get what is wrong if I recently installed the add-on.
 If it was recently, I would likely write a bad review and uninstall it. Yes, it's really a big bummer…
 
 I guess the problem is that I could continue to use the add-on if I had some way to workaround/circumvent the problem at least. Like with a website blacklist.
 I don't want to disable autocorrection at all though.

## Quotation marks

As a developer I'm horribly scared when I do write JSON in the browser and then suddenly my JSON fails to validate (thankfully I did notice that), because it uses these _slighly_ different characters… Aaaaa…

This is similar to the problem below.

## --> and ---, see https://github.com/rugk/unicodify/issues/14

That is so annoying for everywhere I write Markdown. It just breaks Markdown's syntax completely… :see_no_evil: 

That affects e.g. Discourse, GitHub…

## v0.25.4

Version numbers are another problem. No, I don't want them to be replaced…

This is less serious, however, because I at least see the problem immediately and it does not cause some strange JSON parsing errors. 

## Context menu

Actually, you see all these circle around autocompletion. The context menu feature works great, as well… it's very unobstrusive. :smile:

**Comment by tdulcet** [comment]
> Version numbers are another problem. No, I don't want them to be replaced…

This is probably easily fixed by updating the regular expression that matches numbers: https://github.com/rugk/unicodify/blob/8123e09d899cdaa7c31a85299218393ac4a631be/src/content_scripts/autocorrect.js#L306-L307

Note that none of these issues except possibly the version numbers affect Thunderbird, so I think we could release v1.0 on ATN at any time.

I have also noticed a few other issues since we published v0.1 that are not on your above list. I have a PR ready with fixes that I will create as soon as the existing PRs are merged...
